Add some note Components (NoteGrid, NoteGridCollection, and NotePreview)

andres-holguin opened the pull request:
This adds three components that will be used in the generation of notes. NoteGrid and NoteGridCollection serve as containers, while NotePreview is a visual component that shows some Note Data. Styling is subject to change, but the logic of each component is fairly sound. The NoteGridCard component referred to by NoteGridCollection.vue is a WIP, but can be seen behind the preview below.

![image](https://user-images.githubusercontent.com/44207690/87169265-b32dd200-c29d-11ea-9ef5-d0521957e565.png)

andres-holguin committed:
Create NotePreview.vue

Create the component for previewing a Note's data and downloading the note or setting it as a favorite.

andres-holguin committed:
Create NoteGridCollection.vue

Create the component that will generate all the note cards from the array of note data passed to it. Each collection can be treated as a category of notes.

andres-holguin committed:
Create NoteGrid.vue

Create the component that will contain categories/collections of notes and the single preview component.

keelywan commented on src/main/webapp/components/NotePreview.vue:
```diff
@@ -0,0 +1,86 @@
+  }
+</script>
+
+<style>
```
nit: I suggest adding scoped to the style tag in case it interferes with other styling

andres-holguin replied:
Good catch. Unfortunately, the `<v-dialog>` wasn't behaving well to the scoped attribute so I left it out. I will eventually use vuetify components for the .report class so it won't be needed. Not sure what to do about iframe, but I also just tested if this does in fact affect other iframes outside of the component, and it does not. So `scoped` might not even be necessary at all 🤯 .

keelywan replied:
Oh ok that's good to know then!

keelywan reviewed (approved):
Looks good!

andres-holguin committed:
Create template.html

Added a template html file to be used when making new html files in this project.

Aradhya-B reviewed (approved):
Great work Andres, and the template should be super useful for future components!
